stammer.py

```diff
@@ -217,19 +244,22 @@ def process(carrier_path, modulator_path, output_path):
     modulator_bands = make_normalized_bands(modulator_frames, BAND_WIDTH)
 
     print("finding best matches")
-    best_matches = []
+    basis_coefficients = {}
+    best_matches_MATRIX = np.zeros((len(modulator_bands), MAX_BASIS_WIDTH), np.int32) - np.ones((len(modulator_bands), MAX_BASIS_WIDTH), np.int32)
```

Comment:
This leaves an array filled with -1 as a sentry value when assigning matches.
These are indexes into the carrier frames so -1 is a value that should not be reached.
Yes we can use -1  as a reference relative to the end, perhaps, but a loop over the actual indexes in the array would never reach -1.